Makefile

```diff
@@ -0,0 +1,143 @@
+					helpMessage = ""; \
+				} \
+			} else if ($$0 ~ /^[a-zA-Z\-\\_0-9.]+:/) { \
+				helpCommand = substr($$0, 0, index($$0, ":")); \
+				if (helpMessage) { \
+					printf "\033[36m%-20s\033[0m %s\n", \
+						helpCommand, helpMessage; \
+					helpMessage = ""; \
+				} \
+			} else if ($$0 ~ /^##/) { \
+				if (helpMessage) { \
+					helpMessage = helpMessage"\n                     "substr($$0, 3); \
+				} else { \
+					helpMessage = substr($$0, 3); \
+				} \
+			} else { \
+				if (helpMessage) { \
+					print "\n                     "helpMessage"\n" \
+				} \
+				helpMessage = ""; \
+			} \
+		}' \
+		$(MAKEFILE_LIST)
+
+
+MAKE_NPROCS ?= $(shell nproc)
+
+
+## Upgrade the Python requirements
+.PHONY: update-requirements
+update-requirements:
+	pip install --upgrade -r creme/requirements-dev.txt
+
+
+## Upgrade the Python requirements, run the migrations, the creme_populate and generatemedia commands
+.PHONY: update
+update: update-requirements
+	python manage.py migrate
+	python manage.py creme_populate
+	python manage.py generatemedia
+
+
+## Generate the media files
+.PHONY: media
+media:
+	python manage.py generatemedia
+
+
+## Run the Django test suite
+.PHONY: test
+test:
+	python manage.py test --noinput --parallel=${MAKE_NPROCS} $(filter-out $@,$(MAKECMDGOALS))
+
+
+## Run the Django test suite and generate coverage reports
+.PHONY: test-cov
+test-cov:
+	COVERAGE_PROCESS_START=.coveragerc coverage run --source creme/ manage.py test --noinput --parallel=${MAKE_NPROCS} $(filter-out $@,$(MAKECMDGOALS))
+	coverage combine
+	coverage report
+	coverage html
+
+
+## Run the Javascript test suite
+.PHONY: karma
+karma: media
+	node_modules/.bin/karma start .karma.conf.js --browsers=FirefoxHeadless --targets=$(filter-out $@,$(MAKECMDGOALS))
+
+
+## Run the application
+.PHONY: serve
+serve: media
+	python manage.py runserver
+
+
+## Run the Javascript linters
+.PHONY: eslint
+eslint:
+	git diff --name-only origin/master creme/ | { grep '.js$$' || true; } | xargs --no-run-if-empty \
+		node_modules/.bin/eslint \
+			--config .eslintrc \
+			--ignore-path .eslintignore \
+			--format stylish \
+			--quiet
+
+
+## Validates the Python imports with isort
+.PHONY: isort-check
+isort-check:
+	git diff --name-only origin/master creme/ | { grep '.py$$' || true; } | xargs --no-run-if-empty \
+		isort --check --diff --atomic --verbose
+
+
+## Sort the Python imports with isort
+.PHONY: isort-fix
+isort-fix:
+	git diff --name-only origin/master creme/ | { grep '.py$$' || true; } | xargs --no-run-if-empty \
+		isort --atomic --verbose
+
+
+## Validates the Python code with flake8
+.PHONY: flake8
+flake8:
+	flake8 creme
+
+
+## Run all the Python linters
+.PHONY: lint
+lint: isort-check flake8
+
+
+## Collect the messages to translate
+.PHONY: gettext-collect
+gettext-collect:
+	@for appdir in $(shell find creme/ -maxdepth 1 -type d|grep -E 'creme/[^_]+'); do (\
+		cd $${appdir} && \
+		pwd && \
+		django-admin.py makemessages -l fr -i "tests/*" \
```

Comment:
1 - hard coded "fr" is ugly
2 - some exclusion list are more complexe (see creme_core & crudity)

Reply:
For 1) I guess a `LANG?=fr` and run`make LANG=de gettext-collect` will do the job